mathphys/functions.py

```diff
@@ -86,19 +114,80 @@ def save_pickle(data, fname, overwrite=False, makedirs=False):
 def load_pickle(fname):
     """Load ".pickle" file.
 
-    Inputs:
-        fname - filename. May or may not contain the ".pickle" extension.
+    Args:
+        fname (str): Name of the file to load. May or may not contain the
+            ".pickle" extension.
+
+    Returns:
+        data (any builtin type): content of file as a python object.
 
-    Outputs:
-        data - content of file as a python object.
     """
     if not fname.endswith('.pickle'):
         fname += '.pickle'
-    with open(fname, 'rb') as fil:
+
+    func = _gzip.open if is_gzip_file(fname) else open
+
+    with func(fname, 'rb') as fil:
         data = _pickle.load(fil)
     return data
 
 
+def save_hdf5(data, fname, overwrite=False, makedirs=False, compress=False):
+    """Save data to HDF5 file.
+
+    Args:
+        data (any builtin type): python object to be saved
+        fname (str): name of the file to be saved. With or without ".h5".
+        overwrite (bool, optional): whether to overwrite existing file.
+            Defaults to False.
+        makedirs (bool, optional): create dir, if it does not exist.
+            Defaults to False.
+        compress (bool, optional): If True, the arrays in the file will be
+            saved in compressed format, using gzip library. Defaults to False.
```

Comment:
I think this is slightly misleading, in that the compression is implemented by the hdf5 library as a plugin, and not using the python gzip library.

Reply:
Right, I'll try to improve it, thanks!

Reply:
done!